modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java

```diff
@@ -137,6 +137,91 @@ public void setUp() throws Exception {
 
 	@Test
 	public void testAddCustomObjectField() throws Exception {
+		AssertUtils.assertFailure(
+			ObjectFieldBusinessTypeException.class,
+			ObjectFieldConstants.BUSINESS_TYPE_AGGREGATION +
+				" business type is not indexable",
+			() -> ObjectDefinitionTestUtil.addCustomObjectDefinition(
+				false, _objectDefinitionLocalService,
+				Arrays.asList(
+					new AggregationObjectFieldBuilder(
+					).indexed(
+						true
+					).labelMap(
+						LocalizedMapUtil.getLocalizedMap(
+							RandomTestUtil.randomString())
+					).name(
+						"a" + RandomTestUtil.randomString()
+					).objectFieldSettings(
+						Arrays.asList(
+							new ObjectFieldSettingBuilder(
+							).name(
+								"function"
```

Comment:
@p-albuquerque move objectFieldSetting names to `ObjectFieldSettingConstants` and the values that can be moved should be moved as well.

Thanks!